### PR TITLE
Replace maven wtih gradle

### PR DESCRIPTION
**Pull request by utsav00**
## What
- Refer : #143 
- Added gradle and tested locally

### Todo:
- [x] Update build and run steps in docs
- [x] Update docker
- [x] Update github workflows
- [x] Add jacoco reporting
- [x] Remove Maven

## How to run
- `gradle bootRun`

### CODE REVIEW

**Comment by stritti** [comment]
Already great work @utsav00!

**Comment by utsav00** [comment]
Marked as "Ready for Review" to test GH workflow changes. @stritti, can you help me run workflows once I push the changes please?

**Comment by stritti** [comment]
> Marked as "Ready for Review" to test GH workflow changes. @stritti, can you help me run workflows once I push the changes please?

@KzuDemEvin could you assist @utsav00? You updated the GitHub Actions last time I think.

**Comment by KzuDemEvin** [comment]
Awesome work, thanks @utsav00!

In case you did not notice:
I created a pull-request for your repository where I changed some parts or your changes.
PR: https://github.com/utsav00/Diveni/pull/1

This PR fixes the issue that the tests were not running correctly and adds some improvements, changes in the source code.

**Comment by KzuDemEvin** [comment]
The workflow seems to be working for now on.
* I had to make changes to the formatting job. There was an issue when checking out the repository it could not find it, because in this case this pull-request is made from a fork.

* I also had to change the test job. I disabled the [surefire-report](https://github.com/marketplace/actions/surefire-report) step, because it always threw this error: `Resource not accessible by integration`. I have not found a way to fix this error, so the easiest solution was to disable this step for every pull-request from a fork. Only for forks, because it always worked in a normal pull-request.
I would appreciate it if someone could look over my changes.

**Review by KzuDemEvin — APPROVED** [review]

File: backend/build.gradle [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+description = 'Diveni-backend'
+java.sourceCompatibility = JavaVersion.VERSION_11
+
+tasks.withType(JavaCompile) {
+    options.encoding = 'UTF-8'
+}
+
+jar {
+    manifest {
+        attributes 'Main-Class': 'src.main.java.io.diveni.backend.DiveniApplication'
```
Comment (stritti) [inline backend/build.gradle:47]: should this not be without ‘src.main.java‘?

**Review by stritti — CHANGES_REQUESTED** [review]

**Comment by stritti** [comment]
There is currently still a merge conflict to be solved.

File: backend/build.gradle (hunk 1 above)
Comment (KzuDemEvin) [reply backend/build.gradle:47]: Yes, I think you are right. In fact you do not even need this task. If you want to compile your project you call `gradle bootJar`.

That is why I removed it.

File: backend/Dockerfile [hunk 2]
```diff
@@ -1,10 +1,11 @@
...
 WORKDIR /app/Diveni
 COPY src src
-COPY pom.xml .
-RUN mvn package -DskipTests
+RUN gradle build --no-daemon -x test
 
 FROM openjdk:11 as production-stage
 WORKDIR /app/Diveni
-COPY --from=build-stage /app/Diveni/target/app.jar diveni.jar
+RUN echo "Test"
```
[the remark this answers is not in the document]
Comment (KzuDemEvin) [reply backend/Dockerfile]: This is already removed.

**Review by stritti — APPROVED** [review]

**Comment by stritti** [comment]
Great work, @utsav00! Thank you very  much for your support! Also thanks for @KzuDemEvin for assistance.

**Pull request merged**